node/node.go

```diff
@@ -235,3 +174,33 @@ func (n *Node) EventBus() *lltypes.EventBus {
 func (n *Node) ProxyApp() proxy.AppConns {
 	return n.proxyApp
 }
+
+// newTxValidator creates a pubsub validator that uses the node's mempool to check the
+// transaction. If the transaction is valid, then it is added to the mempool
+func newTxValidator(pool mempool.Mempool, poolIDs *mempoolIDs, logger log.Logger) pubsub.Validator {
+	return func(ctx context.Context, i peer.ID, m *pubsub.Message) bool {
+		logger.Debug("transaction of size %d recieved", "bytes", len(m.Data))
+		checkTxResCh := make(chan *abci.Response, 1)
+		err := pool.CheckTx(m.Data, func(resp *abci.Response) {
+			checkTxResCh <- resp
+		}, mempool.TxInfo{
+			SenderID:    poolIDs.GetForPeer(m.GetFrom()),
+			SenderP2PID: corep2p.ID(m.GetFrom()),
+		})
+		switch {
+		case errors.Is(err, mempool.ErrTxInCache):
+			return true
+		case errors.Is(err, mempool.ErrMempoolIsFull{}):
+			return true
+		case errors.Is(err, mempool.ErrTxTooLarge{}):
+			return false
+		case errors.Is(err, mempool.ErrPreCheck{}):
```

Comment:
Orthogonal to this PR but the errors should consistently be vars and wrapped using `fmt.Errorf` and `%w` to decorate them with the necessary info.